### PR TITLE
pipeline changes for docker and openshift cloud

### PR DESCRIPTION
**Pull request by SteffenSeckler**
this moves the pipeline almost entirely to the [openshift cloud](https://en.wikipedia.org/wiki/OpenShift)

**Features:**
* better parallelizability (12 cpus per node, (partially) with hyperthreading
* everything in doxygen containers -- easier testing of different compilers, ... (errors fixed in https://github.com/AutoPas/AutoPas/commit/d59d48950a6c56fb01b0dc3f083b56e09a34a123, https://github.com/AutoPas/AutoPas/commit/b10382677a0279a615bdc4a4bee8ed1b4ced0986 could be identified that way)
* no dependency on installed software, easy "software" installation
* automatic scaling of executors

the only thing that is not in the cloud is the update of the doxygen documentation. As long, as we remain hosting this, this will probably not be changed. (In theory we could move the documentation to github pages)

### CODE REVIEW

**Comment by SteffenSeckler** [comment]
@NeoTheThird for future reference ;)

**Comment by FG-TUM** [comment]
Regarding Doc on github.
I stumbled upon [a gist](https://gist.github.com/vidavidorra/548ffbcdae99d752da02) where someone explained how to deploy a doxygen doc gh-pages. I have neither tested nor fully read it, just leaving it here...

File: Jenkinsfile [hunk 1]
```diff
@@ -10,36 +15,62 @@ pipeline{
...
+                    },
+                    addresssanitizer: {
+                        container('autopas-gcc7-cmake-make') {
+                            dir("build-addresssanitizer"){
+                                sh "cmake -DCMAKE_BUILD_TYPE=Debug -DENABLE_ADDRESS_SANITIZER=ON .."
+                                sh "make -j 4"
+                            }
+                        }
+                    },
+                    addresssanitizerrelease: {
```
Comment (FG-TUM) [inline Jenkinsfile]: Don't want to be nitpicky but camelCase would improve readability here or does this kill jenkis somehow?

**Comment by SteffenSeckler** [comment]
yup sounds good

File: Jenkinsfile (hunk 1 above)
Comment (SteffenSeckler) [reply Jenkinsfile]: yup sounds good

**Pull request merged**